### DISCUSSION TITLE
CTkProgressBar, how to use it.

### DISCUSSION THREAD

**Discussion by sistemasolar5** [TomSchimansky/CustomTkinter#123] [Q&A]
Hi,

I'm new on github, I'm using python, for some projects and I try to implement CTkProgressBar, but it doesn't do what I expect, my code is as follows, I already use the Tk normal progress bar and it works fine, so I don't know if I'm doing this correctly.

Thanks in advance!
```python
i = 0

def progreso():
    for i in range(100):
        win.update()
        progressbar.set(i)
        print(i)
        i += 1

        time.sleep(0.05)

progressbar = customtkinter.CTkProgressBar(master=win,
                                           width=160,
                                           height=20,
                                           border_width=2)
progressbar.place(x=150, y=100)
# Valor del progressBar
```

**Comment by felipetesc** [comment]
Hello There! Is this all your code ? Because:
1 - You need to download and install customtkinter - it's not part of python
2- You need to import customtkinter - so the interpreter can try to locate it and add to the script context
3 - did you created the win instance - python doesn't know what's a window or the win

Can you post the terminal output ?

Wellcome to Github !!

**Reply by sistemasolar5** [reply]
Hi, thank you very much for answer  me, no that is the part of the progress bar only, the rest is working fine, and here is the part on where I'm importint customtkinter, please let me know if you need more information.

Thanks in advance!!!

import customtkinter

customtkinter.set_appearance_mode("dark")# Modes: system (default), light, dark
customtkinter.set_default_color_theme("green")# Themes: blue (default), dark-blue, green

win = customtkinter.CTk()  # create CTk window like the Tk window
# Crea la instacia ventana
#win=Tk()

# Define el tamaño de nuestra ventana
win.geometry("350x250")
